### PR TITLE
tls/store: move sets from basic to buffer, and introduce KeySet.GetFromCertificate()

### PR DESCRIPTION
**Pull request by amery**
moved to avoid circular dependencies 

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `buffer` package for enhanced TLS operations, including utility functions for managing keys and certificates.
	- Added a method to the `KeySet` struct for retrieving private keys associated with certificates.
	- Introduced functions for extracting public keys from private keys and certificates.

- **Refactor**
	- Renamed the package from `basic` to `buffer`, restructuring the organization without altering existing functionalities.

These updates improve the handling of cryptographic materials and enhance error management within the TLS framework.

### CODE REVIEW

File: tls/store/buffer/key_set.go [hunk 1]
```diff
@@ -25,6 +27,24 @@ func (*KeySet) Public(key x509utils.PrivateKey) x509utils.PublicKey {
...
+		return nil, core.Wrap(core.ErrInvalid, "certificate not provided")
+	default:
+		pub, ok := cert.PublicKey.(x509utils.PublicKey)
+		if !ok {
+			return nil, core.Wrap(core.ErrInvalid, "invalid public key")
+		}
+
+		return ks.Get(pub)
+	}
+}
```
[the remark this answers is not in the document]
Comment (amery) [reply tls/store/buffer/key_set.go:46]: the helper you mention doesn't exist but I'll create it

[the remark this answers is not in the document]
Comment (amery) [reply tls/store/buffer/key_set.go]: the x509utils.PublicKey interface is already tested against those implementations

**Review by karasz — APPROVED** [review]
LGTM

**Comment by amery** [comment]
`darvaza.org/x/tls@v0.4.1` tagged

**Pull request merged**